seismic_hazard_analysis/nshm_2010/gmm_hazard.py

```diff
@@ -1,31 +1,28 @@
 from collections.abc import Sequence
-from pathlib import Path
 
 import numpy as np
 import pandas as pd
-from tqdm import tqdm
 
-from empirical.util.classdef import GMM, TectType
-from empirical.util.openquake_wrapper_vectorized import oq_run
+import oq_wrapper as oqw
+from qcore import nhm
 from source_modelling import sources
 
-from .. import hazard, site_source, utils
-from . import utils as nshm2010_utils
+from .. import hazard, utils
+from . import utils as nshm_utils
 
 TECTONIC_TYPE_MAPPING = {
-    "ACTIVE_SHALLOW": TectType.ACTIVE_SHALLOW,
-    "VOLCANIC": TectType.ACTIVE_SHALLOW,
-    "SUBDUCTION_INTERFACE": TectType.SUBDUCTION_INTERFACE,
-    "SUBDUCTION_SLAB": TectType.SUBDUCTION_SLAB,
+    "ACTIVE_SHALLOW": oqw.constants.TectType.ACTIVE_SHALLOW,
+    "VOLCANIC": oqw.constants.TectType.ACTIVE_SHALLOW,
+    "SUBDUCTION_INTERFACE": oqw.constants.TectType.SUBDUCTION_INTERFACE,
+    "SUBDUCTION_SLAB": oqw.constants.TectType.SUBDUCTION_SLAB,
 }
 
 
```

Comment:
Use a typed dict here for clarity:


```suggestion
class SiteProperties(TypedDict):
    vs30: float
    vs30measured: bool
    z1pt0: float
    z2p5: float
    backarc: NotRequired[bool]
```

then change the type in the function signature